### PR TITLE
ADR-786 Check authorisation credentials

### PR DESCRIPTION
**Pull request by davidfes**

### CODE REVIEW

File: app/controllers/actions/AuthorisedAction.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    with FrontendHeaderCarrierProvider
+    with AuthorisedFunctions {
+
+  private val alcoholDutyEnrolment = "HMRC-AD-ORG"
+
+  override def invokeBlock[A](request: Request[A], block: Request[A] => Future[Result]): Future[Result] = {
+    implicit val headerCarrier: HeaderCarrier = hc(request)
+
+    authorised(
+      AuthProviders(GovernmentGateway)
```
Comment (yasaracar) [inline app/controllers/actions/AuthorisedAction.scala:50]: We didn't need to create a new Action, we have the IdentifyAction for this. We are already authorising the users in the IdentifyAction, we just needed to change the code. Please move the relevant code to Identify Action and revert the changes in the controllers.

File: app/controllers/actions/AuthorisedAction.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+class BaseAuthorisedAction @Inject() (
+  appConfig: FrontendAppConfig,
+  override val authConnector: AuthConnector,
+  val parser: BodyParsers.Default
+)(implicit val executionContext: ExecutionContext)
+    extends AuthorisedAction
+    with FrontendHeaderCarrierProvider
+    with AuthorisedFunctions {
+
+  private val alcoholDutyEnrolment = "HMRC-AD-ORG"
```
Comment (yasaracar) [inline app/controllers/actions/AuthorisedAction.scala:44]: Use this from the config please: https://github.com/hmrc/alcohol-duty-returns-frontend/blob/main/conf/application.conf#L107
Anyways the IdentifyAction uses the config value when you move the code. Please see the comment below.

**Review by yasaracar — CHANGES_REQUESTED** [review]

**Comment by davidfes** [comment]
Given changing back the controllers would make a right mess of the commits, I'm going to close and reopen this as a new PR.
